crates/core/src/network/io/reactor.rs

```diff
@@ -398,7 +410,34 @@ impl MetadataVersions {
         }
     }
 
-    pub fn notify(&mut self, header: &Header, connection: &Connection) {
+    pub fn notify_peer_metadata(
+        &mut self,
+        peer: PeerMetadataVersion,
+        connection: &UnboundedConnectionRef,
+    ) {
+        self.update(
+            peer.nodes_config,
+            peer.partition_table,
+            peer.schema,
+            peer.logs,
+        )
+        .into_iter()
+        .for_each(|(kind, version)| {
+            if let Some(version) = version {
+                if version > self.get_latest_version(kind) {
```

Comment:
`notify_observed_version` seems to be doing the same check for the version.

Reply:
Yes, but this one is significantly faster because we are using arcswaps's cache here.